src/evo/providertx.cpp

```diff
@@ -83,10 +93,18 @@ bool CheckProRegTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CValid
     if (ptx.nMode != 0)
         return state.DoS(100, false, REJECT_INVALID, "bad-protx-mode");
 
-    if (ptx.nCollateralIndex >= tx.vout.size())
-        return state.DoS(10, false, REJECT_INVALID, "bad-protx-collateral-index");
-    if (tx.vout[ptx.nCollateralIndex].nValue != 1000 * COIN)
-        return state.DoS(10, false, REJECT_INVALID, "bad-protx-collateral");
+    // when collateralOutpoint refers to an external collateral, we check it further down
+    if (ptx.collateralOutpoint.hash.IsNull()) {
+        if (ptx.collateralOutpoint.n >= tx.vout.size())
+            return state.DoS(10, false, REJECT_INVALID, "bad-protx-collateral-index");
```

Comment:
```suggestion
            return state.DoS(10, false, REJECT_INVALID, "bad-protx-collateral-index");
        }
```

Reply:
Please note that the whole file follows wrong coding style, which means that changes in that file must also follow the same coding style. Better to have a single coding style and everything being wrong than having mixed coding styles.

My plan is still to do a PR which cleans this up in all places related to DIP3. I did many of these mistakes in the past and would like to clean them up all at once.

Reply:
I just feel morally obligated to point them out 😛